test/metabase/public_settings/premium_features_test.clj

```diff
@@ -73,3 +75,112 @@
     ;; for bug reports to come in
     (is (partial= {:valid false, :status "Token does not exist."}
                   (#'premium-features/fetch-token-status* random-fake-token)))))
+
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                          Defenterprise Macro Tests                                             |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(defenterprise greeting
+  "Returns a greeting for a user."
+  metabase-enterprise.util-test
+  [username]
+  (format "Hi %s, you're an OSS customer!" (name username)))
+
+(defenterprise greeting-with-schema :- s/Keyword
+  "Returns a greeting for a user."
+  metabase-enterprise.util-test
+  [username]
+  {username s/Keyword}
+  (format "Hi %s, you're an OSS customer!" (name username)))
+
+(defenterprise greeting-with-valid-token
+  "Returns a non-special greeting for OSS users, and EE users who don't have a valid premium token"
+  metabase-enterprise.util-test
```

Comment:
Another option might be having the syntax for this be

```clj
(defenterprise metabase-enterprise.util-test/greeting-with-valid-token ...)
```

So you don't have to remember to include it for the oss version

Reply:
I tentatively decided against this (see reasoning in the comment below) but could be persuaded otherwise.